it/controllers/TaxCreditsChoiceControllerItSpec.scala

```diff
@@ -0,0 +1,264 @@
+          )
+      )
+
+      server.stubFor(
+        get(urlEqualTo(tcsBrokerUrl))
+          .willReturn(aResponse().withStatus(NOT_FOUND))
+      )
+
+      val request = FakeRequest(GET, url)
+
+      val result = route(app, request)
+
+      result.get.futureValue.header.status mustBe SEE_OTHER
+
+      result.get.futureValue.header.headers.get("Location") mustBe Some("/personal-account/your-address/residential/do-you-live-in-the-uk")
+    }
+
+    List(
+      BAD_REQUEST,
+      IM_A_TEAPOT,
+      INTERNAL_SERVER_ERROR,
+      SERVICE_UNAVAILABLE
+    ).foreach { response =>
+      s"return a SEE_OTHER and redirect to PTA Address Change if the call to TCS fails with a $response" in {
+
+        server.stubFor(
+          get(urlEqualTo(citizenDetailsUrl))
+            .willReturn(ok(FileHelper.loadFile("./it/resources/citizen-details.json")))
+        )
+
+        server.stubFor(
+          get(urlEqualTo(personDetailsUrl))
+            .willReturn(ok(FileHelper.loadFile("./it/resources/person-details.json")))
+        )
+
+        server.stubFor(
+          get(urlPathMatching(s"$cacheMap/.*"))
+            .willReturn(aResponse().withStatus(OK).withBody("""
+                                                              |{
+                                                              |	"id": "session-id",
+                                                              |	"data": {
+                                                              |   "addressPageVisitedDto": {
+                                                              |     "hasVisitedPage": true
+                                                              |   }
+                                                              |	},
+                                                              |	"modifiedDetails": {
+                                                              |		"createdAt": {
+                                                              |			"$date": 1400258561678
+                                                              |		},
+                                                              |		"lastUpdated": {
+                                                              |			"$date": 1400258561675
+                                                              |		}
+                                                              |	}
+                                                              |}
+                                                              |""".stripMargin)
+            )
+        )
+
+        server.stubFor(
+          get(urlEqualTo(tcsBrokerUrl))
+            .willReturn(aResponse().withStatus(response))
+        )
+
+        val request = FakeRequest(GET, url)
+
+        val result = route(app, request)
+
+        result.get.futureValue.header.status mustBe SEE_OTHER
+
+        result.get.futureValue.header.headers.get("Location") mustBe Some("/personal-account/your-address/residential/do-you-live-in-the-uk")
```

Comment:
The view should be a service unavailable page.

Reply:
Done, though I set it to None as there's no header because no onPageLoad route is ran.
It's alright because that's expected, but not sure if you wanted to assert the URL for the error page at all?

Reply:
There is no redirect so the url does not change. A check on the status only is fine here.